Python/thread.c

```diff
@@ -387,6 +387,24 @@ PyThread_delete_key_value(int key)
     PyThread_release_lock(keymutex);
 }
 
+int
+_PyThread_AcquireKeyLock(void)
+{
+    if (keymutex == NULL) {
+        keymutex = PyThread_allocate_lock();
```

Comment:
Maybe we can add a special case if the mutex wasn't created yet: if keymutex is NULL, well, just do nothing, no?

I guess that that the keymutex variable is protected by the GIL.

Reply:
I think this could just be `assert(keymutex != NULL);`.
`keymutex` is first initialized from `Py_Initialize` → `_PyGILState_Init`.